hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/AvroToRowDataConverters.java

```diff
@@ -233,7 +237,11 @@ private static AvroToRowDataConverter createTimestampConverter(int precision) {
               "Unexpected object type for TIMESTAMP logical type. Received: " + avroObject);
         }
       }
-      return TimestampData.fromInstant(instant);
+      if (utcTimezone) {
```

Comment:
Was this not handled for some purpose? The comment in the test said "FlinkOptions.READ_UTC_TIMEZONE doesn't affect in MergeOnReadInputFormat since the option isn't supported in AvroToRowDataConverters". Do we know why it was not supported before?
cc @danny0405

Reply:
> we know why it was not supported before?

Just don't have energy to do that.

Reply:
Got it. Marking resolved as this PR now handles it. Just that we need to have one more test for config set to false as you have mentioned in another comment.